### ISSUE TITLE
Allow for alternative color architectures (specifically base16)

### ISSUE THREAD

**Issue by zeitchef** [cheat/cheat#558]
**Is your feature request related to a problem? Please describe.**
I'm a happy user of "cheat" and have found its author to be a super guy, very responsive, and working hard on this great project. One thing I'd personally love to see is for it to use my terminal's current color configuration. Specifically, the excellent [Base16](https://github.com/chriskempson/base16) project.

**Describe the solution you'd like**
Rather than using a "chroma" theme, I'd like that cheat just use the colors currently provided by the shell. Using the [base16-shell](https://github.com/chriskempson/base16-shell) project, I can easily use a single theme for use in iTerm, Vim or any other CLI. I'd LOVE to be able to view my sheets using this theme as well.

I realize it's probably not a super urgent priority, but it's the ONE thing that keeps me from really enjoying using "cheat" as much as I would like.

**Comment by chrisallenlane** [comment]
Hi, @zeitchef 

Thanks for the kind words. I'm glad to hear that you're enjoying `cheat`!

Regarding your request:

First of all, it's entirely sensible. I'm not afflicted with this personally because I use `solarized-dark` for everything, but I'd be hugely annoyed if I had your problem.

> Rather than using a "chroma" theme, I'd like that cheat just use the colors currently provided by the shell.

Unfortunately, there's some complexity here. There's really no _generalized_ way for `cheat` to use your shell colors. Rather, each color must be assigned on a per-token basis (`keyword`, `variable`, etc) by `chroma`'s parser.

I can think of three possible solutions:

**First and best solution**
While it would not solve the problem in the general case, you might consider opening a pull-request against [chroma][] which implements a `base16` color scheme. Were you to succeed in that, I'd simply need to update `cheat`'s `chroma` dependency, and then two projects would be improved, rather than one :slightly_smiling_face: 

(For what it's worth, I corresponded with `chroma`'s maintainer while refactoring `cheat` into Go, and can state that he's a pleasure to work with and a wonderful maintainer.)

**Second, disgusting solution**
[bat][] appears to support the `base16` color theme. If you have it installed on your system - and if you're only interested in `bash` syntax highlighting - you could create a `cheat` shell function, something like this:

```bash
#!/bin/bash
cheat () {
  cheat $1 | bat --theme=base16 --language=bash --style=plain
}
```
If you were to `source` that function (perhaps in a `.bashrc` file or somewhere), `cheat` would then invoke that function (rather than the executable directly), which would in turn pipe `cheat`'s output through `bat`.

That's gross, but it _might_ work?

**Third solution**
It may also be possible for me to provide users with a mechanism for manually passing color values to `chroma`'s parser. This would probably require some work, but it would solve the problem in the most general case. (I'll look into this when I have some time.)

I hope that's helpful. Feel free to share any feedback here.

[bat]: https://github.com/sharkdp/bat
[chroma]: https://github.com/alecthomas/chroma

**Comment by zeitchef** [comment]
Hi @chrisallenlane,

I took your advice and submitted a pull request to the chroma project, which was just merged. While this solves my immediate issue, I do love the idea of a more robust way of customizing colors for cheat.

Thanks for your work!

**Comment by chrisallenlane** [comment]
Hi, @zeitchef 

That's fantastic. Thanks for making both projects better! :confetti_ball: 

I see that your work was merged on Oct 29, but the current [latest release of `chroma`][release] was published on Sept 22. That being the case, your work won't be available in `cheat` the next version of `chroma` is released.

The `chroma` maintainer is excellent, however, so this likely won't be long. Likewise, I have "dependabot" configured on `cheat/cheat`, so a PR will be submitted automatically once the upstream dependencies are updated.

So, this feature won't be available in `cheat` yet, but it likely will shortly :slightly_smiling_face: 

Thanks!

[release]: https://github.com/alecthomas/chroma/releases/tag/v0.8.1

**Comment by chrisallenlane** [comment]
Tagging as "blocked" while we await the upstream release.

**Comment by chrisallenlane** [comment]
@zeitchef, your changes have been released with `cheat@4.2.0`:
https://github.com/cheat/cheat/releases/tag/4.2.0

Thanks again!
